### PR TITLE
Use "trigger" prefix for fanout & retry metrics

### PR DESCRIPTION
**Pull request by zargarpur**
Currently, the metric names that attempt to be emitted for fanout &
retry use their component names, so they end up being:
- knative.dev/internal/eventing/broker_fanout
- knative.dev/internal/eventing/broker_retry

But we expect the metric names to start with:
- knative.dev/internal/eventing/trigger

### CODE REVIEW

File: pkg/utils/mainhelper/init.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ type InitRes struct {
 // it returns a result object that contains useful artifacts for later use.
 // Unlike sharedmain.Main, Init is meant to be run as a helper function in any main
 // functions, while sharedmain.Main runs controllers with predefined method signatures.
-func Init(component string, opts ...InitOption) (context.Context, *InitRes) {
```
Comment (liu-cong) [inline pkg/utils/mainhelper/init.go:52]: How about making metricNamespace an option, and by default it's the same as `component`? e.g.m Init(component, WithMetricNamespace("trigger"))

Comment (zargarpur) [reply pkg/utils/mainhelper/init.go]: Updated, great suggestion!

File: pkg/utils/mainhelper/options.go [hunk 2]
```diff
@@ -54,6 +55,9 @@ func newInitArgs(component string, opts ...InitOption) initArgs {
 		// Set up signals so we handle the first shutdown signal gracefully.
 		args.ctx = signals.NewContext()
 	}
+	if args.metricNamespace == "" {
```
Comment (liu-cong) [inline pkg/utils/mainhelper/options.go:58]: Can you add a unit test for the new option? Otherwise lgtm

File: cmd/broker/ingress/main.go [hunk 3]
```diff
@@ -63,7 +62,7 @@ func main() {
 		ingress.Port(env.Port),
 		ingress.ProjectID(projectID),
 		metrics.PodName(env.PodName),
-		metrics.ContainerName(containerName),
+		metrics.ContainerName(component),
```
Comment (nachocano) [inline cmd/broker/ingress/main.go:65]: I'd add a comment of why we are doing this...

File: pkg/utils/mainhelper/options.go (hunk 2 above)
Comment (zargarpur) [reply pkg/utils/mainhelper/options.go:58]: Done

File: cmd/broker/ingress/main.go (hunk 3 above)
Comment (zargarpur) [reply cmd/broker/ingress/main.go:65]: I just made this consistent with what the fanout & retry files do, I think ContainerName was a way to include "ingress" as the container name while keeping the metrics working with the "broker" component, but now we can specify the metric name via this PR

**Pull request merged**